src/basins/basins_types.jl

```diff
@@ -0,0 +1,155 @@
+export BasinsOfAttraction, 
+    ArrayBasinsOfAttraction, 
+    SampledBasinsOfAttraction,
+    extract_basins,
+    extract_basins,
+    extract_domain
+    
+#########################################################################################
+# Basins of Attraction structure definition
+#########################################################################################
+"""
+    BasinsOfAttraction{ID}
```

Comment:
Throughout all documentation strings, hide the `ID` information. The user doesn't need to know about it at this stage.